client/src/components/AddNewListDialog/AddNewListDialog.tsx

```diff
@@ -38,12 +40,17 @@ export default function AddNewListDialog({ open, userId, setOpen, setLists, hand
   const [submitting, setSubmitting] = useState<boolean>(false);
   const [titleError, setTitleError] = useState<string>('');
   const [uploading, setUploading] = useState<boolean>(false);
+  const [isPrivate, setIsPrivate] = useState<boolean>(false);
 
   const handleNewListTitleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
     setTitleError('');
     setNewListTitle(event.target.value as string);
   };
 
+  const handlePrivacyChange = () => {
+    setIsPrivate(!isPrivate);
```

Comment:
```suggestion
    setIsPrivate(prevPrivate => !prevPrivate);
```

use prev state so that it's not dependant on current state (react does batch updates so this is a much more safe way to toggle states) to prevent race conditions